CommandLine/emake/EnigmaCallbacks.cpp

```diff
@@ -38,7 +38,7 @@ const LogMessage& CallBack::GetNextLogMessage(bool &end) const {
   ++logIt;
   if (logIt == logMessages.end()) {
     end = true;
-    return LogMessage();
+    return std::move(LogMessage());
```

Comment:
@JoshDreamland is this safe too? I know it fixes the warnings and that my calling code does expect that when `end == true` that the returned reference is invalid. So is this an ok fix?

Reply:
It's always safe to move an object, but when you do that, consider the result a swap, or better, just assume the object you moved is now default.

Reply:
Yay, so then this is fine, that's just how I want it to work.

Reply:
@rpjohnst looked over this and said this is unsafe and that you probably didn't see I am returning the reference. This can lead to undefined behavior so I am just going to change the functions to return by value.

Reply:
Ok: bc007c52f0c158c6bf275a88d31b31ce65476a75

I've changed some functions to return by value, performance isn't much of a concern because the compiler can do return value optimizations.

I've tested that the output continues to work and according to Rusky it will protect against undefined behavior. This issue is resolved now.